contracts/modules/cover/Cover.sol

```diff
@@ -781,12 +800,13 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     return uint(activeCover[assetId].totalActiveCoverInAsset);
   }
 
-  function isPoolAllowed(uint productId, uint poolId) external view returns (bool) {
+  // Returns true if the pool exists and is allowed to have the product
+  function isPoolAllowed(uint productId, uint poolId) public view returns (bool) {
```

Comment:
Do we want to check for deprecated products here? Cover purchases will be blocked for the product, so not sure if it's worth the extra gas.